fern/docs/pages/user-guide/etcd-metadata-exchange.md

```
@@ -0,0 +1,320 @@
+
+Fetching retrieves a remote agent's metadata from etcd by looking up `{namespace}/{remote_agent}/{metadata_label}`. If the key exists, the metadata is loaded immediately. If the key does not yet exist, NIXL sets up an `etcd::Watcher` on the key and waits for it to appear, with a configurable timeout (`etcdWatchTimeout`).
+
+After successfully fetching metadata, NIXL sets up a persistent watcher on the remote agent's prefix key for invalidation notifications (see [Watcher Component](#watcher-component) below).
+
+<CodeBlocks>
+```python title="Python"
+# Fetch a remote agent's metadata from ETCD
+# (no IP/port arguments = ETCD mode)
+agent.fetch_remote_metadata("remote_agent_name")
+
+# Verify the metadata is loaded
```

markdownlint-cli2 (0.23.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)